### PR TITLE
[bazel] Refactor the mask ROM signature verification code to make it easier to re-bazelize

### PR DESCRIPTION
**Pull request by mcy**
This PR:
- Moves all sigverify code into its own directory.
- Splits `mod_exp.h` into two files.
- Fixes up the Meson rules.
- Adds new Bazel rules that accurately capture how all these things actually depend on each other.

Child of https://github.com/lowRISC/opentitan/pull/11847; only the last four commits are relevant, and the first two are just renames.

### CODE REVIEW

**Comment by drewmacrae** [comment]
Oh no, the Mask ROM Build and Test step wasn't designed to let you move tests. It definitely should work if you get all the Mask ROM tests passing. I was going to work out a strategy to intentionally drop tests on Monday that should help.

**Comment by drewmacrae** [comment]
It should work fine if you haven't broken any tests and you fix the target in [mask-rom-tests.sh](https://cs.opensource.google/opentitan/opentitan/+/master:ci/scripts/mask-rom-tests.sh;drc=af9be146c5bb650176b4b23972d37fea9f07149f;l=13)

File: sw/device/silicon_creator/lib/sigverify/BUILD [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    ],
+)
+
+opentitan_functest(
+    name = "mod_exp_ibex_functest",
+    srcs = ["mod_exp_ibex_functest.c"],
+    verilator = verilator_params(
+        timeout = "long",
+        tags = [
+            "manual",
```
Comment (drewmacrae) [inline sw/device/silicon_creator/lib/sigverify/BUILD:67]: Can you put a todo for the manual tests linking https://github.com/lowRISC/opentitan/issues/11860 IDK if we'll get to it but it's nice to know why it's different.

**Review by drewmacrae — APPROVED** [review]
Yeah this all seems reasonable. You'll have to fix the test targets for CI, but then it should work fine.

File: sw/device/silicon_creator/lib/sigverify/BUILD [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+)
+
+opentitan_functest(
+    name = "sigverify_dynamic_functest",
+    srcs = ["sigverify_dynamic_functest.c"],
+    verilator = verilator_params(
+        timeout = "long",
+        tags = [
+            "manual",
+            "verilator",
```
Comment (timothytrippel) [inline sw/device/silicon_creator/lib/sigverify/BUILD:211]: you shouldn't need this as it is a required tag that gets automatically added [here](https://github.com/lowRISC/opentitan/blob/5852b71c5c0d70709699b4a18fc6de921bc34836/rules/opentitan_test.bzl#L140). You will need a `cpu:X` tag though **_if you want one_**, as that is not a required tag, [rather a default](https://github.com/lowRISC/opentitan/blob/5852b71c5c0d70709699b4a18fc6de921bc34836/rules/opentitan_test.bzl#L100).

File: sw/device/silicon_creator/lib/sigverify/BUILD [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+)
+
+opentitan_functest(
+    name = "sigverify_functest",
+    srcs = ["sigverify_functest.c"],
+    verilator = verilator_params(
+        timeout = "long",
+        tags = [
+            "manual",
+            "verilator",
```
Comment (timothytrippel) [inline sw/device/silicon_creator/lib/sigverify/BUILD:194]: shouldn't need this, see below

File: sw/device/silicon_creator/lib/sigverify/BUILD [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+)
+
+opentitan_functest(
+    name = "mod_exp_otbn_functest",
+    srcs = ["mod_exp_otbn_functest.c"],
+    verilator = verilator_params(
+        timeout = "long",
+        tags = [
+            "manual",
+            "verilator",
```
Comment (timothytrippel) [inline sw/device/silicon_creator/lib/sigverify/BUILD:121]: shouldn't need this, see below.

File: sw/device/silicon_creator/lib/sigverify/BUILD [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+)
+
+opentitan_functest(
+    name = "mod_exp_ibex_functest",
+    srcs = ["mod_exp_ibex_functest.c"],
+    verilator = verilator_params(
+        timeout = "long",
+        tags = [
+            "manual",
+            "verilator",
```
Comment (timothytrippel) [inline sw/device/silicon_creator/lib/sigverify/BUILD:68]: see below

File: sw/device/silicon_creator/lib/sigverify/BUILD (hunk 2 above)
Comment (timothytrippel) [reply sw/device/silicon_creator/lib/sigverify/BUILD:211]: (this was my fault, I refactored the `verilator_params()` function and did not refactor the invocations to it; sorry about that. I will make a pass through the codebase for these once this merges)

**Review by timothytrippel — APPROVED** [review]

File: sw/device/silicon_creator/lib/sigverify/BUILD (hunk 2 above)
Comment (mcy) [reply sw/device/silicon_creator/lib/sigverify/BUILD:211]: As discussed offline, let's do this as a separate cleanup in the name of maintaining velocity (i.e., CI being painfully slow).

**Comment by alphan** [comment]
post-merge lgtm

**Pull request merged**